File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/LetterDB/Letter/LetterColumns.cs
using CorrespondenceSystem.Modules.Enums.Letter;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.LetterDB.Columns;

[ColumnsScript("LetterDB.Letter")]
[BasedOnRow(typeof(LetterRow), CheckNames = true)]
[AlignRight]
public class LetterColumns
{
    [EditLink, QuickFilter]
    public string Title { get; set; }
    [QuickFilter]
    public string Tag { get; set; }
    [QuickFilter]
    public string GrandSubjectTitle { get; set; }
    [QuickFilter]
    public string LetterNo { get; set; }
    [QuickFilter]
    public DateTime CreatedDate { get; set; }
    public States State { get; set; }
    public LetterTypes LetterType { get; set; }

}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/LetterDB/LetterColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { States } from "../Modules/Enums.Letter.States";
import { LetterRow } from "./LetterRow";

export interface LetterColumns {
    Title: Column<LetterRow>;
    Tag: Column<LetterRow>;
    GrandSubjectTitle: Column<LetterRow>;
    LetterNo: Column<LetterRow>;
    CreatedDate: Column<LetterRow>;
    State: Column<LetterRow>;
    LetterType: Column<LetterRow>;
}

export class LetterColumns extends ColumnsBase<LetterRow> {
    static readonly columnsKey = 'LetterDB.Letter';
    static readonly Fields = fieldsProxy<LetterColumns>();
}

[States]; // referenced types